### PR TITLE
chore: document usage of hashed passwords

### PR DESCRIPTION
**Pull request by baurmatt**
We didn't like to store the password in plain text and tried to specify the password hash. It worked. So here is some documentation and an e2e test :)

### CODE REVIEW

**Comment by baurmatt** [comment]
DCO added.

**Review by gbartolini — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

File: docs/src/declarative_role_management.md [hunk 1]
```diff
@@ -159,6 +159,23 @@ never expires, mirroring the behavior of PostgreSQL. Specifically:
     had been previously created. In 1.20.0, doing this might inadvertently
     result in setting existing passwords to `NULL`.
 
+### Password hashed
+
+You can also provide pre-encrypted passwords by specifiy the password in MD5/SCRAM-SHA-256 hash format:
```
Comment (gbartolini) [inline docs/src/declarative_role_management.md:164]:
```suggestion
You can also provide pre-encrypted passwords by specifying the password
in MD5/SCRAM-SHA-256 hash format:
```

**Review by gbartolini — APPROVED** [review]

**Comment by gbartolini** [comment]
@baurmatt can you please rebase?

**Comment by baurmatt** [comment]
@gbartolini Applied your suggestion and rebased. Thanks! :)

**Review by leonardoce — APPROVED** [review]

**Pull request merged**